### PR TITLE
[DOC-307] [2024.1] Replace/decommision a node

### PR DESCRIPTION
**Pull request by aishwarya24**
Deploy preview - https://deploy-preview-22207--infallible-bardeen-164bc9.netlify.app/preview/yugabyte-platform/manage-deployments/remove-nodes/
To do - replace recommission screenshot @rajmaddy89 (needs better quality), https://deploy-preview-22207--infallible-bardeen-164bc9.netlify.app/preview/yugabyte-platform/manage-deployments/remove-nodes/#recover-a-node (update the screenshot in this section)
Add diagram

### CODE REVIEW

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 1]
```diff
@@ -16,6 +16,40 @@ menu:
...
+## Replace a live or unreachable node
+
+If a node (live or unreachable) needs to be replaced, do the following:
+
+1. Navigate to **Universes**, select your universe, and open the **Nodes** tab.
+1. Click the associated node **Actions > Replace Node**.
+
+    ![Replace Node Actions](/images/ee/replace-node.png)
+
+Clicking the **Replace Node** button opens a popup window to confirm the replacement of this node. Confirming the task redirects to the task page and will take a few minutes to complete.
```
Comment (charleswang234) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:28]: I don't think confirming replacing redirects to task page, it should reload the page and the loading circle and the progress bar should slowly run

@rajmaddy89  can you confirm?

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 2]
```diff
@@ -16,6 +16,40 @@ menu:
...
+
+For Infrastructure as a service (IaaS) such as AWS and GCP, YugabyteDB Anywhere returns the node back to the IaaS. For on-premises universes, the unresponsive node will most likely have failed to clean up its data directory, home directory, and more. So it will be placed in a Decommissioned state in the on-premises provider which prevents the node from being added to a new universe.
+
+### Check on-premises node state
+
+The on-premises node instances have three states: In use, Free, and Decommissioned and the following illustration describes the workflow of the states.
+
+![Decommissioned node workflow](/images/ee/decommissioned-node.png)
```
Comment (charleswang234) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:36]: We can remove the "Action: Decommission", as we do not support this currently

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 3]
```diff
@@ -16,6 +16,40 @@ menu:
...
+To check the state of an on-premises node, navigate to **Configs > On-Premises Datacenters**, then click its associated on-premises configuration, and click **Instances**.
```
Comment (charleswang234) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:37]: Would be good to have a picture for this

cc @rajmaddy89

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md (hunk 2 above)
Comment (charleswang234) [reply docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:36]: Also instead of release node in the "Release node: data cleanup fails" and "Release node:  cleanup succeeds" let's change "Release node" to "Replace node"

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 4]
```diff
@@ -16,6 +16,40 @@ menu:
...
+
+### Recommission on-premises node instance
+
+A node in "Decommissioned" state can be placed back into the "Free" node state pool after it becomes reachable and is cleaned up.
+
+Perform the following steps to recommission a node.
+
+1. For a node instance in the Decommissioned state, navigate to **Configs** > On-Premises Datacenters**, then click its associated on-premises configuration, and click **Instances**.
```
Comment (charleswang234) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:45]: After "On-Premises Datacenters" UI shows "**" which is incorrect

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 5]
```diff
@@ -16,6 +16,40 @@ menu:
...
+
+1. On the Instances page, click **Actions > Recommission Node**.
+
+    ![Recommission Node](/images/ee/recommission-node.png)
+
+Clicking the **Recommission Node** button opens a popup window to confirm the recommission of this node. Confirming the task redirects to the task page and will take a few minutes to complete.
+
 ## Eliminate an unresponsive node
```
Comment (charleswang234) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:53]: We should say now that replace is the recommended flow for elinating an unreachable node. 

And that now remove, release, delete, add are not recommended and are put in the advanced actions

Comment (charleswang234) [reply docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:53]: Will need pics for this

cc @rajmaddy89

Comment (charleswang234) [reply docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:53]: In this case, we can put the start/stop node above remove, release, add, delete, as it will be used more often

**Review by charleswang234 — CHANGES_REQUESTED** [review]
Add some comments, thanks

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 6]
```diff
@@ -16,6 +16,40 @@ menu:
...
+
+### Recommission on-premises node instance
+
+A node in "Decommissioned" state can be placed back into the "Free" node state pool after it becomes reachable and is cleaned up.
+
+Perform the following steps to recommission a node.
+
+1. For a node instance in the Decommissioned state, navigate to **Configs** > On-Premises Datacenters**, then click its associated on-premises configuration, and click **Instances**.
+
+1. On the Instances page, click **Actions > Recommission Node**.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:47]:
```suggestion
1. Under Instances, for the decommissioned node, click **Actions > Recommission Node**.
```

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 7]
```diff
@@ -16,6 +16,40 @@ menu:
...
+
+    ![Recommission Node](/images/ee/recommission-node.png)
+
+Clicking the **Recommission Node** button opens a popup window to confirm the recommission of this node. Confirming the task redirects to the task page and will take a few minutes to complete.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:51]:
```suggestion
1. Click OK to confirm.

YugabyteDB Anywhere starts the node recommissioning process, and you can view the progress on the **Tasks** tab.
```

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md (hunk 1 above)
Comment (aishwarya24) [reply docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:28]: I have just added this instead and I think this should suffice - 
1. Click OK to confirm.
YugabyteDB Anywhere (YBA) starts the node replacement process, and you can view the progress on the **Tasks** tab.

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md (hunk 5 above)
Comment (aishwarya24) [reply docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:53]: @charleswang234 - If its not recommended, then should those sequence of steps just be removed? For what situations would a user actually choose it ? 

At the moment ill comment it out and recommend using replace.

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 8]
```diff
@@ -16,13 +16,55 @@ menu:
 type: docs
 ---
 
+## Replace a live or unreachable node
+
+If a node (live or unreachable) needs to be replaced, do the following:
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:21]:
```suggestion
To replace a live node for extended maintenance or replace an unhealthy node, do the following: 
```

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 9]
```diff
@@ -16,13 +16,55 @@ menu:
...
+
+    ![Replace Node Actions](/images/ee/replace-node.png)
+
+1. Click OK to confirm.
+
+YugabyteDB Anywhere (YBA) starts the node replacement process, and you can view the progress on the **Tasks** tab.
+
+For cloud providers (AWS, Azure, or GCP), YBA returns the node back to the provider.
+
+For on-premises universes, because the unresponsive node will likely fail to clean up its data directory, home directory, and so on, YBA sets the state to Decommissioned. This prevents the node from being added to a new universe.
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:34]:
```suggestion
For on-premises universes, clean up of existing data directories and running processes may fail if the node is unhealthy. In such cases, YBA sets the state to Decommissioned. This prevents the node from being added to a new universe.
```

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 10]
```diff
@@ -16,13 +16,55 @@ menu:
...
+
+### Recommission a decommissioned on-premises node
+
+You can return a Decommissioned node to the on-premises provider Free pool after it becomes reachable and is cleaned up.
+
+Perform the following steps to recommission a node:
+
+1. Navigate to **Configs > Infrastructure > On-Premises Datacenters**, select the associated on-premises configuration, and click **Instances**.
+
+1. Under Instances, for the decommissioned node, click **Actions > Recommission Node**.
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:51]:
```suggestion
1. Under Instances, for the decommissioned node, click **Actions > Recommission Node**. YBA will now re-attempt to clean up existing data directories and processes on this node.
```

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 11]
```diff
@@ -16,13 +16,55 @@ menu:
...
+YugabyteDB Anywhere (YBA) starts the node recommissioning process, and you can view the progress on the **Tasks** tab.
+
 ## Eliminate an unresponsive node
 
 If a virtual machine or a physical server in a universe reaches its end of life and has unrecoverable hardware or other system issues (such as problems with its operating system, disk, and so on) it is detected and displayed in the YugabyteDB Anywhere UI as an unreachable node, as per the following illustration:
 
 ![Unreachable Node Actions](/images/ee/node-actions-unreachable.png)
 
-When this happens, new Master leaders are elected for the underlying data shards. Because the universe enters a partially under-replicated state, it would not be able to tolerate additional failures. To remedy the situation, you can eliminate the unreachable node by taking actions in the following sequence:
+When this happens, new Master leaders are elected for the underlying data shards. Because the universe enters a partially under-replicated state, it would not be able to tolerate additional failures. To remedy the situation, it is recommended that you [replace the node](#replace-a-live-or-unreachable-node).
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:65]:
```suggestion
When this happens, new tablet leaders are elected for the underlying data shards and availability is maintained. If no other nodes are available to rebuild the lost copies of these tablets, the universe enters a partially under-replicated state and it would not be able to tolerate additional failures. To remedy the situation, it is recommended that you [replace the node](#replace-a-live-or-unreachable-node).
```

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 12]
```diff
@@ -36,9 +78,27 @@ A node status displayed in the UI is not always entirely indicative of the node'
 
 {{< /note >}}
 
+## Start and stop node processes
+
+### Stop a process
+
+If a node needs the intervention, you can click its associated **Actions > Stop Processes**.
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:85]:
```suggestion
If a node needs to be briefly taken out of service (for example, to perform a quick OS patch), you can click its associated **Actions > Stop Processes**. It is expected that this node will be returned to service soon through the **Actions > Start Processes** operation.
```

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 13]
```diff
@@ -36,9 +78,27 @@ A node status displayed in the UI is not always entirely indicative of the node'
...
+
+After the YB-TServer and (where applicable) YB-Master server are stopped, the node status is updated and the instance is ready for the planned system changes.
+
+Generally, when a YB-Master is stopped on a node, YugabyteDB Anywhere automatically attempts to start a new YB-Master on another node in the same Availability Zone as the node on which YB-Master is stopped. This ensures that the number of YB-Master servers equals the replication factor (RF) and YB-Master servers are never underreplicated.
+
+It is recommended not to stop more than (RF - 1) / 2 processes at any given time. For example, on an RF=3 cluster with three nodes, there can only be one node with stopped processes to allow the majority of the nodes to perform Raft consensus operations.
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:91]: @charleswang234 : Should we rephrase this to say we don't recommend stopping multiple nodes at the same time because they can contain copies of the same tablet?

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 14]
```diff
@@ -36,9 +78,27 @@ A node status displayed in the UI is not always entirely indicative of the node'
...
+
+### Start a process
+
+You can restart the node's processes by navigating to **Universes**, selecting your universe, then selecting **Nodes**, and then clicking **Actions > Start Processes** corresponding to the node. This returns the node to the Live state.
+
+In some cases, the system might experience an unrecoverable error. To mitigate, you can use the **Release Instance** option for the stopped node. This also removes the backing instance. For details, see [Remove a node](#remove-node).
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:97]: @charleswang234 : In the context of start process, we don't routinely expect unrecoverable errors right? Do we need this part?

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md (hunk 13 above)
Comment (charleswang234) [reply docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:91]: Yes, we should recommend that we should not stop multiple nodes at the same time (especially if in the same az).

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md (hunk 14 above)
Comment (charleswang234) [reply docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:97]: I don't think we need line 97 anymore, since now we only allow retries on the same task for node ops

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md (hunk 5 above)
Comment (charleswang234) [reply docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:59]: Currently we have remove, release, delete, add for old users, as they are used to that workflow. Other than that, it should not be used.

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 15]
```diff
@@ -36,9 +78,27 @@ A node status displayed in the UI is not always entirely indicative of the node'
...
+
+### Stop a process
+
+If a node needs to be briefly taken out of service (for example, to perform a quick OS patch), you can click its associated **Actions > Stop Processes**. It is expected that this node will be returned to service soon through the **Actions > Start Processes** operation.
+
+After the YB-TServer and (where applicable) YB-Master server are stopped, the node status is updated and the instance is ready for the planned system changes.
+
+Generally, when a YB-Master is stopped on a node, YugabyteDB Anywhere automatically attempts to start a new YB-Master on another node in the same Availability Zone as the node on which YB-Master is stopped. This ensures that the number of YB-Master servers equals the replication factor (RF) and YB-Master servers are never underreplicated.
+
+It is recommended not to stop more than (RF - 1) / 2 processes at any given time. For example, on an RF=3 cluster with three nodes, there can only be one node with stopped processes to allow the majority of the nodes to perform Raft consensus operations.
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:91]:
```suggestion
In general, it is not recommended to execute the "Stop Processes" action on multiple nodes simultaneously. For example, two stopped nodes might share a common tablet. This could cause unavailability on a universe with replication factor 3.
```

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 16]
```diff
@@ -36,9 +78,27 @@ A node status displayed in the UI is not always entirely indicative of the node'
...
+
+### Start a process
+
+You can restart the node's processes by navigating to **Universes**, selecting your universe, then selecting **Nodes**, and then clicking **Actions > Start Processes** corresponding to the node. This returns the node to the Live state.
+
+In some cases, the system might experience an unrecoverable error. To mitigate, you can use the **Release Instance** option for the stopped node. This also removes the backing instance. For details, see [Remove a node](#remove-node).
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:97]:
```suggestion
```

**Review by iSignal — APPROVED** [review]
Incorporating feedback from @charleswang234

File: docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md [hunk 17]
```diff
@@ -16,13 +16,55 @@ menu:
...
+1. Navigate to **Universes**, select your universe, and open the **Nodes** tab.
+1. Click the associated node **Actions > Replace Node**.
+
+    ![Replace Node Actions](/images/ee/replace-node.png)
+
+1. Click OK to confirm.
+
+YugabyteDB Anywhere (YBA) starts the node replacement process, and you can view the progress on the **Tasks** tab. As part of the node replacement process, all data (tablets) on the existing node will be moved to other nodes to ensure that the desired replication factor is maintained throughout the operation.
+
+For cloud providers (AWS, Azure, or GCP), YBA returns the node back to the provider.
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/manage-deployments/remove-nodes.md:32]:
```suggestion
For cloud providers (AWS, Azure, or GCP), YBA returns the existing node back to the provider and provisions a new replacement node from the cloud provider. For on-premises universes, the existing node is returned to the [on-premises provider node pool](../configure-yugabyte-platform/on-premises-nodes/) and a new replacement node is selected from the free pool.
```

**Comment by aishwarya24** [comment]
Created PR - https://github.com/yugabyte/yugabyte-db/pull/22595
